src/unify/mod.rs

```diff
@@ -236,6 +236,8 @@ impl<K: UnifyKey> VarValue<K> {
 // Option<U>:LatticeValue.
 
 impl<S: UnificationStore> UnificationTable<S> {
+    // FIXME(eddyb) (not backwards-compat) this should
+    // use a `Default` bound on `S` instead.
```

Comment:
I'd rather just have `UnificationStore` require `Default`. It's not a trait that others outside the crate are meant to implement.